clase 1 19.4/src/main/java/services/App.java

```diff
@@ -0,0 +1,34 @@
+package services;
+
+import scanner.IScanner;
+import scanner.ScannerClass;
+import scanner.impl.MenuSizeImpl;
+import services.command.Command;
+import services.command.InitializeMediumDifficultyCommandImpl;
+import services.command.InitializeHardDifficultyCommandImpl;
+import java.util.ArrayList;
+import java.util.List;
+
+public class App {
+    public void run(){
+
+        MenuSizeImpl menuSizeImpl = new MenuSizeImpl();
+        menuSizeImpl.add("Please enter the difficulty you want to play with: 1-Medium, 2-Hard: ");
+        menuSizeImpl.add("You have entered the number: ");
+        IScanner scanner = new ScannerClass();
+        int numberDecided = scanner.askUserForWayOfPlaying(menuSizeImpl);
```

Comment:
As you are using inheritance, Why does the child class need to send it to the parent?, because the parent is the ScannerClass , and the menuSizeImpl extends from ScannerClass, so, in this line: 
`scanner.askUserForWayOfPlaying(menuSizeImpl);` 
you are sending the child to the father's method.

Reply:
in the father class i make the methods. And i use the child class to pass the information to the interface. Later the class that implements the intefaces prints the information passed in the child class